### PR TITLE
CTest: skip tests on return code 77

### PR DESCRIPTION
**Pull request by dparanic**
If test's return code is 77, then this test is marked as skipped.
This return code is used as de facto practice. The GNU standard approach in case test is skipped is to exit the program with code 77.

### CODE REVIEW

**Comment by capatober** [comment]
Could you please clarify why this code was chosen?

File: test/rng_testsuite/conformance_tests/minstd_rand_rand0_test.pass.cpp [hunk 1]
```diff
@@ -27,6 +27,9 @@
 //     The 10000th consecutive invocation of a default-constructed object of type minstd_rand
 //     produces the value 399268537
 
+#include <oneapi/dpl/execution>
+#include "support/pstl_test_config.h"
```
Comment (paveldyakov) [inline test/rng_testsuite/conformance_tests/minstd_rand_rand0_test.pass.cpp:31]: Does it make sense to rename pstl_test_config.h to test_config.h? As it is also included in RNG tests

File: test/rng_testsuite/conformance_tests/minstd_rand_rand0_test.pass.cpp [hunk 2]
```diff
@@ -27,6 +27,9 @@
 //     The 10000th consecutive invocation of a default-constructed object of type minstd_rand
 //     produces the value 399268537
 
+#include <oneapi/dpl/execution>
```
Comment (paveldyakov) [inline test/rng_testsuite/conformance_tests/minstd_rand_rand0_test.pass.cpp:30]: Could you please tell - why do we need this include here?

Comment (dparanic) [reply test/rng_testsuite/conformance_tests/minstd_rand_rand0_test.pass.cpp]: There were failures on Linux with gcc 9.3.0, this include was fixing this failures. But now I found better solution. So this change is deprecated.

File: test/rng_testsuite/conformance_tests/minstd_rand_rand0_test.pass.cpp (hunk 1 above)
Comment (dparanic) [reply test/rng_testsuite/conformance_tests/minstd_rand_rand0_test.pass.cpp]: Yes, this make sense. I have created issue.

**Review by MikeDvorskiy — CHANGES_REQUESTED** [review]
Essentially, we don't need to total change the all test files  (~ 40)...
I believe  just TestUtils::done() function should be changed.

**Review by capatober — APPROVED** [review]

File: test/support/utils.h [hunk 3]
```diff
@@ -792,14 +795,16 @@ transform_reduce_serial(InputIterator first, InputIterator last, T init, BinaryO
...
+int
+done(bool is_passed = true)
 {
-#if _PSTL_TEST_SUCCESSFUL_KEYWORD
-    return "done";
-#else
-    return "passed";
-#endif
+    if(is_passed){
+        ::std::cout <<"Passed\n";
```
Comment (MikeDvorskiy) [inline test/support/utils.h:802]: you missed "done"

File: test/support/utils.h [hunk 4]
```diff
@@ -792,14 +795,16 @@ transform_reduce_serial(InputIterator first, InputIterator last, T init, BinaryO
...
+        return 0;
+    }else{
+        ::std::cout <<"Skipped\n";
```
Comment (MikeDvorskiy) [inline test/support/utils.h:805]: I suggest writing additional info, f.e.,  "<id> group tests are skipped" or something like that.

File: test/support/utils.h [hunk 5]
```diff
@@ -792,14 +795,16 @@ transform_reduce_serial(InputIterator first, InputIterator last, T init, BinaryO
     return init;
 }
 
-static const char*
-done()
+int
+done(bool is_passed = true)
```
Comment (MikeDvorskiy) [inline test/support/utils.h:799]: a caller passes an integer (macro value)..  :"done" is accepting "bool"...  please, arrange types

Comment (MikeDvorskiy) [inline test/support/utils.h:799]: I guess name of argument should be "is_done" or revert logic  - "is_skipped"

File: test/support/utils.h (hunk 3 above)
Comment (dparanic) [reply test/support/utils.h:802]: Should we use specify one output on success? For rng_testsuite tests we use Passed and for other tests we use done.

**Review by paveldyakov — DISMISSED** [review]

File: test/support/utils.h (hunk 4 above)
Comment (dparanic) [reply test/support/utils.h:805]: When we run tests with cmake, it prints all skipped tests. If we run one test other way, then we know, why it is skipped. Could you describe case when we need additional info?

File: test/support/utils.h (hunk 5 above)
Comment (dparanic) [reply test/support/utils.h]: Applied

Comment (dparanic) [reply test/support/utils.h]: Done

File: test/support/utils.h (hunk 3 above)
Comment (dparanic) [reply test/support/utils.h]: Applied

File: test/support/utils.h [hunk 6]
```diff
@@ -792,14 +795,20 @@ transform_reduce_serial(InputIterator first, InputIterator last, T init, BinaryO
...
+    if(is_done == 1){
 #if _PSTL_TEST_SUCCESSFUL_KEYWORD
-    return "done";
+        ::std::cout << "done\n";
 #else
-    return "passed";
+        ::std::cout << "passed\n";
 #endif
+        return 0;
+    }else{
```
Comment (MikeDvorskiy) [inline test/support/utils.h:808]: According to our code style:
```
...
}
else
{
...
```

File: test/support/utils.h [hunk 7]
```diff
@@ -792,14 +795,20 @@ transform_reduce_serial(InputIterator first, InputIterator last, T init, BinaryO
     return init;
 }
 
-static const char*
-done()
+int
+done(int is_done = 1)
 {
+    if(is_done == 1){
```
Comment (MikeDvorskiy) [inline test/support/utils.h:801]: According to our code style:
```
if(..)
{
....
}
else
{
..
}
```

File: test/support/utils.h (hunk 4 above)
Comment (dparanic) [reply test/support/utils.h:812]: Discussed it and decided to keep it in mind on future updates.

**Review by MikeDvorskiy — APPROVED** [review]

**Pull request merged**